packages/atomic/src/components/result-template-components/atomic-result-printable-uri/atomic-result-printable-uri.tsx

```diff
@@ -0,0 +1,110 @@
+import {Component, h, Prop, State} from '@stencil/core';
+import {Result} from '@coveo/headless';
+import {ResultContext} from '../result-template-decorators';
+import {parseXML} from '../../../utils/utils';
+import {
+  Bindings,
+  BindStateToI18n,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+
+/**
+ * The ResultUri component displays the URI, or path, to access a result.
+ * @part result-printable-uri-list - The result printable uri list
+ * @part result-printable-uri-list-expanded - The expanded result printable uri list
+ * @part result-printable-uri-list-element - A result printable uri list element
+ * @part result-printable-uri-link - A result printable uri clickable link
+ * @part result-printable-uri-list-ellipsis - The clickable ellipsis of a result uri
+ */
+@Component({
+  tag: 'atomic-result-printable-uri',
+  styleUrl: 'atomic-result-printable-uri.pcss',
+  shadow: false,
+})
+export class AtomicResultPrintableUri {
+  @InitializeBindings() public bindings!: Bindings;
+  @ResultContext() private result!: Result;
+  private minNumOfParts = 3;
+
+  @State() listExpanded = false;
+  @BindStateToI18n()
+  @State()
+  private strings = {
+    collapsedUriParts: () => this.bindings.i18n.t('collapsedUriParts'),
+  };
+  @State() error!: Error;
+
+  /**
+   * The maximum number of Uri parts to display, has to be over the minimum of `3` in order to be effective. Putting `0` will disable the ellipsis.
+   */
+  @Prop() maxNumOfParts = 5;
+
+  private renderEllipsis() {
+    return (
+      <li part="result-printable-uri-list-element">
+        <button
+          part="result-printable-uri-list-ellipsis"
+          aria-label={this.strings.collapsedUriParts()}
+          onClick={() => (this.listExpanded = true)}
+        >
+          ...
+        </button>
+      </li>
+    );
+  }
+
+  private get allParents() {
+    return Array.from(
+      parseXML(this.result.raw.parents!).getElementsByTagName('parent')
+    ).map((parent) => {
+      const name = parent.getAttribute('name');
+      const uri = parent.getAttribute('uri')!;
+      return (
+        <li part="result-printable-uri-list-element">
+          <a part="result-printable-uri-link" href={uri}>
+            {name}
+          </a>
+        </li>
+      );
+    });
+  }
+
+  private renderParents() {
+    const parents = this.allParents;
+    if (
+      this.listExpanded ||
+      this.maxNumOfParts === 0 ||
+      parents.length <= Math.max(this.minNumOfParts, this.maxNumOfParts)
```

Comment:
I think it would be better to validate the number using Stencil's `@watch` prop, so that any code using `maxNumOfParts` will not need to repeat the Math.max validation.

The `minNumOfParts` could be scoped inside the validation function too,

Reply:
No watch doesn't run on initialization but on later changes, but yes for the rest :)